sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3023,9 +3023,27 @@ class Analyzer(
   object ResolveAlterTableChanges extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
       case a @ AlterTable(_, _, t: NamedRelation, changes) if t.resolved =>
+        // Keeps track of new columns being added. Keys are normalized parent names and
+        // values are field names that belong to their parent.
```

Comment:
Can we be more clear here? For example, if we add a column `a.b.c.d`, which part should be the key? `a.b.c` or `a.b`?